### DISCUSSION TITLE
Why no `.map`?

### DISCUSSION THREAD

**Discussion by bmulholland** [cypress-io/cypress#24016] [Questions and Help]
Cypress includes no way to map over a list of elements. It's possible to do so, but only with a very roundabout approach: creating an array, doing `.each`, adding each to the array, then returning it at the end. `map` is such a typical primitive that its lack of inclusion must be a fundamental choice -- one that isn't documented, as far as I could find. There's not even a discussion about it here already, which makes me feel like I must be missing something incredibly obvious.

**Why isn't there a map? What are we expected to do instead? Why isn't this explained in the documentation?**

Specifically: I have a list of elements I am operating on, and I need to store their IDs in an alias for an assertion later. The IDs are stored in `data-testid`. I get the list of elements, and now I want each attribute -- but `invoke` (silently!) returns only the attribute of the first item. I have to somehow loop over them, and the above workaround is the only choice. [There are StackOverflows about this](https://stackoverflow.com/questions/57080643/how-to-get-array-of-specific-attributes-values-in-cypress), of course, but the code is fighting so much against the structure of Cypress that something feels off. This is the second time I've run into this obstacle, and each time it's taken me over an hour of banging my head against the wall.

**Comment by jjhelguero** [comment]
[Lodash](https://docs.cypress.io/api/utilities/_) is bundled with cypress which includes `.map()`. Have you taken a look at that?

**Reply by bmulholland** [reply]
I did try to use that: it was confusing, poorly documented, and required contortions that broke my head trying to use it. I got lost in a maze of `Cypress._.chain` and `Cypress.dom.unwrap` and something similar about arrays. My attempts resulted in nothing but more confusion, an error about mixing synchronous and asynchronous code, and a further certainty that I'm missing something.

In my experience, when you have to fight a tool so much, it's usually hard for good reason. Doubly so for something so common and basic that I've already hit it twice in my tiny ~20 test suite. Hence my question.

**Reply by jjhelguero** [reply]
I may not understand your situation completely, however in instances I have used lodash, its either to set a variable outside of cy commands or within a `.then()` to yield a new subject.
```js
// variable
const randomNumber = Cypress._.random(2e10)

// .then()
cy.get('.selectors')
   .then( ($el) => Cypress._.map($el, "innerText"))
```
